controllers/secret_cert_controller.go

```diff
@@ -177,7 +177,44 @@ func (r *SecretCertReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 					log.Error(err, "unable to update ApiDef")
 				}
 
-				log.Info("api def updated succesfully")
+				log.Info("api def updated successfully")
+
+				return ctrl.Result{}, nil
+			}
+		}
+
+		for domain := range apiDefList.Items[idx].Spec.PinnedPublicKeysSecretNames {
+			if desired.Name == apiDefList.Items[idx].Spec.PinnedPublicKeysSecretNames[domain] {
+				certID, err := klient.Universal.Certificate().Upload(ctx, tlsKey, tlsCrt)
+				if err != nil {
+					return ctrl.Result{Requeue: true}, err
+				}
+
+				if apiDefList.Items[idx].Spec.PinnedPublicKeys == nil {
+					apiDefList.Items[idx].Spec.PinnedPublicKeys = map[string]string{}
+				}
+
+				apiDefList.Items[idx].Spec.PinnedPublicKeys[domain] = certID
+
+				err = r.Update(ctx, &apiDefList.Items[idx])
+
+				if apierrors.IsConflict(err) {
+					// The Pod has been updated since we read it.
+					// Requeue to try to reconciliate again.
+					return ctrl.Result{Requeue: true}, nil
+				}
+
+				if apierrors.IsNotFound(err) {
+					// The Pod has been deleted since we read it.
```

Comment:
```suggestion
					// The Pod has been deleted since we read it.
```

Should it be APIDefiniton/resource instead of Pod?